coldfront/plugins/qumulo/tasks.py

```diff
@@ -163,3 +167,157 @@ def __validate_results(quota_usages, logger) -> bool:
         logger.warn("Unsuccessful ingestion of quota daily usage. Check the results.")
 
     return success
+
+
+class ResetAcl(object):
+    qumulo_api = None
+    sub_allocations = None
+
+    def __init__(self, allocation: Allocation):
+        self.allocation = allocation
+        self.debug = ENV.bool('DEBUG', default=False)
+        self.fs_path = allocation.get_attribute(name='storage_filesystem_path')
+        self.is_allocation_root = QumuloAPI.is_allocation_root_path(
+            self.fs_path
+        )
+        access_allocations = AclAllocations.get_access_allocations(allocation)
+        self.rw_group = AclAllocations.get_allocation_rwro_group_name(
+            access_allocations,
+            'rw'
+        )
+        self.ro_group = AclAllocations.get_allocation_rwro_group_name(
+            access_allocations,
+            'ro'
+        )
+        self.reset_exclude_paths = []
+        if self.is_allocation_root:
+            self.sub_allocations = []
+            links = None
+            try:
+                links = AllocationLinkage.objects.get(parent=allocation)
+            except AllocationLinkage.DoesNotExist:
+                pass
+            if links is not None:
+                for child in links.children.all():
+                    self.reset_exclude_paths.append(
+                        child \
+                            .get_attribute(name='storage_filesystem_path') \
+                            .rstrip('/')
+                    )
+                    self.sub_allocations.append(child)
+        else:
+            self._setup_qumulo_api()
+            self.parent_aces = AclAllocations.get_sub_allocation_parent_aces(
+                allocation,
+                self.qumulo_api
+            )
+
+    # <debugging functions>
+    def __log_acl(self, msg):
+        global logger
+        if not self.debug:
+            return None
+        logger.warn(msg)
+
+    def __log_acl_reset(self, path):
+        self.__log_acl(f'Resetting "directory content" ACLs for path: {path}')
+    # </debugging functions>
+
+    def _setup_qumulo_api(self):
```

Comment:
@ris-bmulligan what are the use cases when the QUMULO API is not required? Perhaps, this is telling that this class has too many responsibilities. A sub-class with more responsibilities or a decorator could separate the concerns.

Reply:
I added that because the task would throw errors about lower-level classes/objects not being "pickle"-able.  Evidently that is related to multi-threading (really async processing) or multi-processing.  I managed to get past that by making the Qumulo API object instantiation be a "check if exists, else instantiate" operation.